clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/cache/AzureServerGroupCachingAgent.groovy

```diff
@@ -85,7 +85,45 @@ class AzureServerGroupCachingAgent extends AzureCachingAgent {
       it.attributes.processedCount = (it.attributes.processedCount ?: 0) + 1
     }
 
-    result
+    if (result.cacheResults[AZURE_SERVER_GROUPS.ns]) {
+      result
+    } else {
+      // run the cache cleanup routine on an empty server group list only for now
```

Comment:
The "dead/stale" server group entries are added to the eviction list so if the "cacheResult" collection contains at least one non-empty list we successfully process the eviction list and the dead entries are removed. It's only for the case where the "cacheResult" collection is made of empty lists that we fail to properly evict/delete stale cache entries (this issue might be in the general code - still looking to see what's causing it).